pilot/pkg/bootstrap/server.go

```diff
@@ -601,29 +601,65 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 			Time:    args.KeepaliveOptions.Time,
 			Timeout: args.KeepaliveOptions.Timeout,
 		})
-		msgSizeOption := grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(args.MCPMaxMessageSize))
-		conn, err := grpc.DialContext(ctx, configSource.Address, securityOption, msgSizeOption, keepaliveOption)
-		if err != nil {
-			log.Errorf("Unable to dial MCP Server %q: %v", configSource.Address, err)
-			cancel()
-			return err
+
+		s.mcpOptions = &coredatamodel.Options{
+			DomainSuffix: args.Config.ControllerOptions.DomainSuffix,
 		}
 
-		mcpController := coredatamodel.NewController(options)
+		mcpController := coredatamodel.NewController(s.mcpOptions)
 		sinkOptions := &sink.Options{
 			CollectionOptions: collections,
 			Updater:           mcpController,
 			ID:                clientNodeID,
 			Reporter:          reporter,
 		}
 
+		msgSizeOption := grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(args.MCPMaxMessageSize))
+		conn, err := grpc.DialContext(ctx, configSource.Address, securityOption, msgSizeOption, keepaliveOption)
+		if err != nil {
+			log.Errorf("Unable to dial MCP Server %q: %v", configSource.Address, err)
+			cancel()
+			return err
+		}
 		cl := mcpapi.NewResourceSourceClient(conn)
 		mcpClient := sink.NewClient(cl, sinkOptions)
 		configz.Register(mcpClient)
 		clients = append(clients, mcpClient)
-
 		conns = append(conns, conn)
 		configStores = append(configStores, mcpController)
+
+		// create MCP IncrementalController
+		for _, addr := range args.IncrementalMCPSrvAddrs {
+			s.incrementalMcpOptions = &coredatamodel.Options{
+				DomainSuffix: args.Config.ControllerOptions.DomainSuffix,
+			}
+
+			s.mcpDiscovery = coredatamodel.NewMCPDiscovery()
```

Comment:
This doesn't seem right ... this code would overwrite the mcpDiscovery server for each new MCP server.

Reply:
Totally agree, didn't mean to place the creation of MCPDiscovery in the loop.